enhancement(policy, aws): Construct CIDR rules for referenced AWS security groups by querying associated network interfaces (#27071)
cilium-team/cilium: Documentation/security/aws.rst

***********************************************

This document serves as an introduction to using Cilium to enforce policies

> based on AWS metadata. It is a detailed walk-through of getting a single-node
> Cilium environment running on your machine. It is designed to take 15-30 minutes
> with some experience running Kubernetes.

Reviewer: Edits for clarity
```suggestion
based on AWS metadata. It provides a detailed walk-through of running a single-node
Cilium environment on your machine. It is designed to take 15-30 minutes
for users with some experience running Kubernetes.
```

Cilium's AWS Metadata filtering capability enables explicit whitelisting
of communication between a subset of pods (identified by Kubernetes labels)
with a set of destination EC2 ENIs (identified by membership in an AWS security group).

> In this example, the destination EC2 elastic network interfaces are attached to
> EC2 instances that are members of a single AWS security group ('sg-0f2146100a88d03c3')
> and pods with label class=xwing should only be able to make connections outside the
> cluster to the destination network interfaces in that security group.

Reviewer: Use clear, plain language, formatting
```suggestion
In this example, the destination EC2 elastic network interfaces are attached to
EC2 instances that are members of a single AWS security group ('sg-0f2146100a88d03c3').
Pods with label ``class=xwing`` should only be able to make connections outside the
cluster to the destination network interfaces in that security group.
```

            securityGroupsIds:
            - 'sg-0f2146100a88d03c3'

> This policy allows traffic from pod *xwing* to any AWS elastic network interface
> that is in the security group with ID ``sg-0f2146100a88d03c3``.

Reviewer: Edits for clarity
```suggestion
This policy allows traffic from pod *xwing* to any AWS elastic network interface
in the security group with ID ``sg-0f2146100a88d03c3``.
```

Validate that derived policy is in place
----------------------------------------

Every time that a new policy with ToGroups rules is added, an equivalent policy
(also called "derivative policy"), will be created. This policy will contain the
set of CIDRs that correspond to the specification in ToGroups, e.g., the IPs of
all network interfaces that are part of a specified security group. The list of

> IPs will be updated periodically.

Reviewer: Use present tense
```suggestion
IPs is updated periodically.
```
